karate-core/src/main/java/com/intuit/karate/http/ApacheHttpClient.java

```diff
@@ -200,7 +213,24 @@ public Response invoke(HttpRequest request) {
         this.request = request;
         RequestBuilder requestBuilder = RequestBuilder.create(request.getMethod()).setUri(request.getUrl());
         if (request.getHeaders() != null) {
-            request.getHeaders().forEach((k, vals) -> vals.forEach(v -> requestBuilder.addHeader(k, v)));
+            //request.getHeaders().forEach((k, vals) -> vals.forEach(v -> requestBuilder.addHeader(k, v)));
+                request.getHeaders().forEach((k, vals) ->
```

Comment:
I'd like to create a util method somewhere, but thanks for this, I am ok to check this in and then refactor

I tried something here: `com.intuit.karate.http.Request.getCookies()` and the same "pattern" is needed in `karate-mock-servlet` - again I can spend some time to think, e.g. use a lambda so that the "final" conversion from Netty Cookie to Apache (or Servlet) cookie can be done with less lines of code

Reply:
one more thing, we need to have a test for this in `karate-core`, I just made a change to enable this: https://github.com/intuit/karate/commit/fcde0ad83fb5f61f95ad87a08148290a387f0d7d - see if you can add one or two - I like this pattern where a single junit test method can test the whole "flow" but with just a few lines of code - I am hoping I can create something similar for UI tests also :)

Reply:
actually it may make more sense to have the extra cookies tests in `KarateHttpMockHandlerTest` but the above is good to have

Reply:
@ptrthomas - sure thing, will have some updates later in the day today. Thank you.

Reply:
> I'd like to create a util method somewhere, but thanks for this, I am ok to check this in and then refactor
> 
> I tried something here: `com.intuit.karate.http.Request.getCookies()` and the same "pattern" is needed in `karate-mock-servlet` - again I can spend some time to think, e.g. use a lambda so that the "final" conversion from Netty Cookie to Apache (or Servlet) cookie can be done with less lines of code

I created a small util - hopefully this will be ok.

Reply:
> one more thing, we need to have a test for this in `karate-core`, I just made a change to enable this: [fcde0ad](https://github.com/intuit/karate/commit/fcde0ad83fb5f61f95ad87a08148290a387f0d7d) - see if you can add one or two - I like this pattern where a single junit test method can test the whole "flow" but with just a few lines of code - I am hoping I can create something similar for UI tests also :)

added a test.

Reply:
> actually it may make more sense to have the extra cookies tests in `KarateHttpMockHandlerTest` but the above is good to have

I noticed KarateMockHandlerTest has some foul test cases around cookies -for ex `testCookieWithDateInThePast` and also noticed some of the code to take care-off expired cookies etc is gone now. Please let me know if you want me to revisit.

Reply:
as I think I said earlier, apologies for removing the expired cookies checks, it struck me that as a testing framework, users may want to submit expired cookies deliberately to the server - so we should allow that control - also I had a LOT of trouble with cookies not working for the mock-server

anyway, see this commit - I added a test for cookies, and as far as I can tell cookies are being sent correctly: https://github.com/intuit/karate/commit/7755b90adada57648f67f719f813d5befeb662dc

Reply:
sure thing - I was inly thinking of those expired cookies etc & yeah they flow through just fine.

anythimg else I should focus on for this PR other then merging that methodInvoker change? Thank you

Reply:
great. about this PR - it is become confusing for me and I have made the method invoker changes already - actually it is removed etc. so let us close this and start a fresh one if needed

Reply:
got it! I thought we needed cookiestore changes for Apache Client - no worries, will close this one and go from there. Thank you.